### PR TITLE
BUG: Avoid AmbiguousTimeError on groupby

### LINKED ISSUE #14682: AmbiguousTimeError on groupby when including a DST change
#### A small, complete example of the issue

```python
#!/usr/bin/env python
import pandas as pd
df=pd.DataFrame([1477786980,1477790580],columns=['ts'])
df['date']=pd.to_datetime(df.ts, unit='s').dt.tz_localize('UTC').dt.tz_convert('Europe/Madrid')
df.set_index('date', inplace=True)

dfo = df.groupby(pd.TimeGrouper('5min'))
```
#### Expected Output
```
                           ts
date                         
2016-10-30 02:20:00+02:00   1
2016-10-30 02:25:00+02:00   0
2016-10-30 02:30:00+02:00   0
2016-10-30 02:35:00+02:00   0
2016-10-30 02:40:00+02:00   0
2016-10-30 02:45:00+02:00   0
2016-10-30 02:50:00+02:00   0
2016-10-30 02:55:00+02:00   0
2016-10-30 02:00:00+01:00   0
2016-10-30 02:05:00+01:00   0
2016-10-30 02:10:00+01:00   0
2016-10-30 02:15:00+01:00   0
2016-10-30 02:20:00+01:00   1
```

#### Output of ``pd.show_versions()``

<details>
# Paste the output here pd.show_versions() here
>>> pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.12.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-47-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: None.None

pandas: 0.19.1
nose: 1.3.7
pip: 9.0.1
setuptools: 28.6.1
Cython: 0.25.1
numpy: 1.11.2
scipy: None
statsmodels: None
xarray: None
IPython: None
sphinx: 1.4.8
patsy: None
dateutil: 2.4.2
pytz: 2016.7
blosc: None
bottleneck: None
tables: None
numexpr: 2.6.1
matplotlib: None
openpyxl: 2.2.6
xlrd: None
xlwt: None
xlsxwriter: None
lxml: 3.5.0
bs4: 4.4.1
html5lib: 0.999
httplib2: 0.9.1
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None
pandas_datareader: None
</details>

The above code raises an `AmbiguousTimeError` exception, when grouping by a time-date series including a DST change. In the above example the unix timestamps are for the recent DST change in Europe.

The stack trace is:
```
Traceback (most recent call last):
  File "./t.py", line 7, in <module>
    dfo = df.groupby(pd.TimeGrouper('5min'))
  File "/usr/local/lib/python2.7/dist-packages/pandas/core/generic.py", line 3984, in groupby
    **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pandas/core/groupby.py", line 1501, in groupby
    return klass(obj, by, **kwds)
  File "/usr/local/lib/python2.7/dist-packages/pandas/core/groupby.py", line 370, in __init__
    mutated=self.mutated)
  File "/usr/local/lib/python2.7/dist-packages/pandas/core/groupby.py", line 2382, in _get_grouper
    binner, grouper, obj = key._get_grouper(obj)
  File "/usr/local/lib/python2.7/dist-packages/pandas/tseries/resample.py", line 1062, in _get_grouper
    r._set_binner()
  File "/usr/local/lib/python2.7/dist-packages/pandas/tseries/resample.py", line 237, in _set_binner
    self.binner, self.grouper = self._get_binner()
  File "/usr/local/lib/python2.7/dist-packages/pandas/tseries/resample.py", line 245, in _get_binner
    binner, bins, binlabels = self._get_binner_for_time()
  File "/usr/local/lib/python2.7/dist-packages/pandas/tseries/resample.py", line 660, in _get_binner_for_time
    return self.groupby._get_time_bins(self.ax)
  File "/usr/local/lib/python2.7/dist-packages/pandas/tseries/resample.py", line 1118, in _get_time_bins
    base=self.base)
  File "/usr/local/lib/python2.7/dist-packages/pandas/tseries/resample.py", line 1262, in _get_range_edges
    closed=closed, base=base)
  File "/usr/local/lib/python2.7/dist-packages/pandas/tseries/resample.py", line 1326, in _adjust_dates_anchored
    return (Timestamp(fresult).tz_localize(first_tzinfo),
  File "pandas/tslib.pyx", line 621, in pandas.tslib.Timestamp.tz_localize (pandas/tslib.c:13694)
  File "pandas/tslib.pyx", line 4308, in pandas.tslib.tz_localize_to_utc (pandas/tslib.c:74816)
pytz.exceptions.AmbiguousTimeError: Cannot infer dst time from Timestamp('2016-10-30 02:20:00'), try using the 'ambiguous' argument
```

Code works if the series does not include a DST change (e.g. one day earlier):

```python
#!/usr/bin/env python
import pandas as pd
df=pd.DataFrame([1477700580,1477704180],columns=['ts'])
df['date']=pd.to_datetime(df.ts, unit='s').dt.tz_localize('UTC').dt.tz_convert('Europe/Madrid')
df.set_index('date', inplace=True)

dfo = df.groupby(pd.TimeGrouper('5min'))

print dfo.count()
```

gets:

```
                           ts
date                         
2016-10-29 02:20:00+02:00   1
2016-10-29 02:25:00+02:00   0
2016-10-29 02:30:00+02:00   0
2016-10-29 02:35:00+02:00   0
2016-10-29 02:40:00+02:00   0
2016-10-29 02:45:00+02:00   0
2016-10-29 02:50:00+02:00   0
2016-10-29 02:55:00+02:00   0
2016-10-29 03:00:00+02:00   0
2016-10-29 03:05:00+02:00   0
2016-10-29 03:10:00+02:00   0
2016-10-29 03:15:00+02:00   0
2016-10-29 03:20:00+02:00   1
```

### PR DESCRIPTION
**Pull request by j-santander**
closes #14682

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Could you add your example from the issue as a test? Also a release note in the whatsnew file.

**Comment by j-santander** [comment]
it seems I've broken something.... I'll try to fix.

File: pandas/tseries/resample.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 import pandas as pd
 from pandas.core.base import AbstractMethodError, GroupByMixin
+from pandas.core.config_init import pc_ambiguous_as_wide_doc
```
Comment (jreback) [inline pandas/tseries/resample.py]: why are you importing this?

File: pandas/tseries/tests/test_resample.py [hunk 2]
```diff
@@ -1912,7 +1912,18 @@ def test_resample_size(self):
         right = Series(val, index=ix)
         assert_series_equal(left, right)
 
-    def test_resmaple_dst_anchor(self):
+    def test_resample_across_dst(self):
+        #14682
+        df1 = DataFrame([1477786980, 1477790580], columns=['ts'])
```
Comment (jreback) [inline pandas/tseries/tests/test_resample.py]: give a 1-liner about what you are testing here

File: pandas/tseries/resample.py (hunk 1 above)
Comment (j-santander) [inline pandas/tseries/resample.py]: IDE mischieve, I'll remove.

File: pandas/tseries/tests/test_resample.py (hunk 2 above)
Comment (j-santander) [inline pandas/tseries/tests/test_resample.py]: I will.

File: pandas/tseries/resample.py [hunk 3]
```diff
@@ -1284,8 +1285,13 @@ def _adjust_dates_anchored(first, last, offset, closed='right', base=0):
     # See https://github.com/pandas-dev/pandas/issues/8683
 
     first_tzinfo = first.tzinfo
+    last_tzinfo = last.tzinfo
```
Comment (jreback) [inline pandas/tseries/resample.py]: put the issue number as a comment and a brief expl. future readers will appreciate!

File: pandas/tseries/index.py [hunk 4]
```diff
@@ -439,7 +439,7 @@ def _generate(cls, start, end, periods, name, offset,
                 tz = tz.localize(date.replace(tzinfo=None)).tzinfo
 
         if tz is not None and inferred_tz is not None:
-            if not inferred_tz == tz:
+            if not tslib.get_timezone(inferred_tz) == tslib.get_timezone(tz):
```
Comment (jreback) [inline pandas/tseries/index.py]: hmm, this should not be necessary; str comparison already does this

File: pandas/tseries/tests/test_resample.py (hunk 2 above)
Comment (jreback) [inline pandas/tseries/tests/test_resample.py]: this is a bug, so you can put in 0.19.2

File: pandas/tseries/resample.py (hunk 3 above)
Comment (j-santander) [inline pandas/tseries/resample.py]: I will

File: pandas/tseries/index.py (hunk 4 above)
Comment (jreback) [inline pandas/tseries/index.py]: hmm, can you repro that directly via a DatetimeIndex construction call?

does it break anything else?

Comment (j-santander) [inline pandas/tseries/index.py]: ok, I've traced the issue to the tz of input index:

I do:

```python
df1 = DataFrame([1477786980, 1477790580], columns=['ts'])
dti1 = DatetimeIndex(pd.to_datetime(df1.ts, unit='s').dt.tz_localize('UTC').dt.tz_convert('Europe/Madrid'))
```

Now, this dti1 index has a _funny_ tz attribute: `<DstTzInfo 'Europe/Paris' LMT+0:09:00 STD>`
When resampling, the index tz is used in creating the new index and it is where the assert comes.

That probably means I don't need this change... but the fix is needed elsewhere.

Comment (j-santander) [inline pandas/tseries/index.py]: Sorry, my bad.... To reproduce:
```python
t1=Timestamp(2016,10,30,02,23).tz_localize('Europe/Madrid',ambiguous=True)
t2=Timestamp(2016,10,30,02,13).tz_localize('Europe/Madrid',ambiguous=False)
dti=DatetimeIndex(start=t1,end=t2,freq='H',tz='Europe/Madrid')
```

I've also found that the same check is done in pandas.tseries.tools.py:

```python
def _infer_tzinfo(start, end):
    def _infer(a, b):
        tz = a.tzinfo
        if b and b.tzinfo:
            if not (tslib.get_timezone(tz) == tslib.get_timezone(b.tzinfo)):
                raise AssertionError('Inputs must both have the same timezone,'
                                     ' {0} != {1}'.format(tz, b.tzinfo))
        return tz

    tz = None
    if start is not None:
        tz = _infer(start, end)
    elif end is not None:
        tz = _infer(end, start)
    return tz
```

I'll submit now the changes and see if this time they are ok.

File: pandas/tseries/tests/test_resample.py [hunk 5]
```diff
@@ -1912,7 +1912,31 @@ def test_resample_size(self):
...
+
+        # The expected DatetimeIndex after resampling.
+        # 2016-10-30 02:00:00+02:00, 2016-10-30 02:00:00+01:00
+        df2 = DataFrame([1477785600, 1477789200], columns=['ts'])
+        dti2 = DatetimeIndex(pd.to_datetime(df2.ts, unit='s')
+                             .dt.tz_localize('UTC')
+                             .dt.tz_convert('Europe/Madrid'))
+        df = DataFrame([5, 5], index=dti1)
+        assert_frame_equal(
+            df.resample(rule='H').sum(),
```
Comment (jreback) [inline pandas/tseries/tests/test_resample.py]: do 
```
result = df.resample(....)
expected = ....
assert_frame_equal(result, expected)
```

**Comment by jreback** [comment]
lgtm.

cc @rockg

File: doc/source/whatsnew/v0.19.2.txt [hunk 6]
```diff
@@ -43,3 +43,4 @@ Bug Fixes
 
 
 - Explicit check in ``to_stata`` and ``StataWriter`` for out-of-range values when writing doubles (:issue:`14618`)
+- Fix AmbiguousTimeError exception when resampling a DatetimeIndex in local TZ, covering a DST change (:issue:`14682`)
```
Comment (j-santander) [inline doc/source/whatsnew/v0.19.2.txt]: I'll change

File: pandas/tseries/tests/test_resample.py (hunk 5 above)
Comment (j-santander) [inline pandas/tseries/tests/test_resample.py]: ok

**Comment by rockg** [comment]
Looks fine.

**Comment by jreback** [comment]
thanks @j-santander 

keep em' coming!

**Pull request closed without merging**